Document alternate install methods (#620)
phylum-dev/cli: docs/command_line_tool/alternate_install.md

---
title: Alternate Installation Methods

> category: 6255e67693d5200013b1fa41

Reviewer: ```suggestion
category: 6255e67693d5200013b1fa3e
```
Given that the location of this file changed from the `docs/knowledge_base` directory to the `docs/command_line_tool` directory, the assumption is that this page should fall under the `COMMAND LINE TOOL` category.

   ```sh
   cargo install --path cli
   ```

> ## Install with phylum-ci

Reviewer: ```suggestion
## Install with Python
```

> See the [phylum-ci](https://github.com/phylum-dev/phylum-ci) project for details.

Reviewer: ```suggestion
The [`phylum` Python package](https://pypi.org/project/phylum/) provides a script entry point, `phylum-init`, for bootstrapping the Phylum CLI.
It requires a modern version of Python. See the [phylum-ci](https://github.com/phylum-dev/phylum-ci) project for full detail.
```

Author: > It requires a modern version of Python.

I'm rejecting this note because Python 2.x has been end-of-life for longer than Phylum has been a company.

```sh

> pipx install phylum-ci
> phylum-init

Reviewer: ```suggestion
pip install phylum
phylum-init
```
The repository is named `phylum-ci`. The package is named `phylum`. The script entry point is named `phylum-init`. Not confusing at all 😛.
Even though I greatly prefer `pipx` for installing packages that provide script entry points, that assumes one more pre-requisite for the user. Sticking with plain old `pip` is best for this form of the documentation. More detail is provided in the links to the package/repo, provided above.

Reviewer 2: Is Python stuff automatically added to the $PATH? Or does that require manual steps.

Reviewer: The short answer is yes.

The script entry points will be automatically available on the `$PATH`, whether the install was with `pip` or `pipx`. With a `pip` based install, if the package was installed in a Python virtual environment, then that environment needs to be active for the scripts to be on the `$PATH`.

Author: > Even though I greatly prefer `pipx` for installing packages that provide script entry points, that assumes one more pre-requisite for the user. Sticking with plain old `pip` is best for this form of the documentation. More detail is provided in the links to the package/repo, provided above, for intrepid users.

This is the alternate install page, so we are already dealing with "intrepid users". And `pipx` has significant advantages here. I still believe `pipx` deserves the recommendation on this page

Author: > Is Python stuff automatically added to the $PATH? Or does that require manual steps.

This is one of the benefits of `pipx` since it will handle this. But even with `pip`, there should be a warning printed if the script directory isn't in your path when the package is installed.
